modules/hostap/src/supp_api.c

```diff
@@ -476,22 +495,10 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 	strncpy(ssid_null_terminated, params->ssid, WIFI_SSID_MAX_LEN);
 	ssid_null_terminated[params->ssid_length] = '\0';
 
-	if (!wpa_cli_cmd_v("set_network %d ssid \"%s\"",
-			   resp.network_id, ssid_null_terminated)) {
-		goto out;
-	}
-
-	if (!wpa_cli_cmd_v("set_network %d scan_ssid 1", resp.network_id)) {
-		goto out;
-	}
-
-	if (!wpa_cli_cmd_v("set_network %d key_mgmt NONE", resp.network_id)) {
-		goto out;
-	}
-
-	if (!wpa_cli_cmd_v("set_network %d ieee80211w 0", resp.network_id)) {
-		goto out;
-	}
+	wpa_cli_cmd("set_network %d ssid \"%s\"", resp.network_id, ssid_null_terminated);
```

Comment:
Just my opinion, but TBH I'm not really convinced such macrobatics leads to improved code readability, especially if macros have no indication of side effects, this just looks like a regular function call. Perhaps it's just my taste, but I usually find plain C easier to follow, even if there's more code to read.

Maybe as an alternative, we could just stack those multiple command calls into a single `if` statement to avoid code bloat, something like:
```c
	if (!wpa_cli_cmd_v("set_network %d ssid \"%s\"",  resp.network_id, ssid_null_terminated) ||
	    !wpa_cli_cmd_v("set_network %d scan_ssid 1", resp.network_id) ||
	    !wpa_cli_cmd_v("set_network %d key_mgmt NONE", resp.network_id) ||
	    !wpa_cli_cmd_v("set_network %d ieee80211w 0", resp.network_id)) {
		goto out;
	}
```

Reply:
I think we could do the checks like this. The ugliness in the code is mostly in couple of functions which have several macro calls in a row and this would solve it nicely without any extra macrobatics.

Reply:
I am all for macrobatics :). 

> Describe alternatives you've considered
> We could leave it as is or switch from macro to function that can take an array of strings?

How about something like this? It still is like macro, but batches the command execution and does it at once with better error handling?

```
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdarg.h>

bool zephyr_wpa_cli_cmd_v(const char *cmd, ...);
void wpa_printf(int level, const char *fmt, ...);

#define MSG_ERROR 0
#define MAX_COMMANDS 100
#define MAX_COMMAND_LENGTH 256

bool execute_wpa_cli_commands(char commands[][MAX_COMMAND_LENGTH]) {
    bool status = true;
    int command_count = 0;

    // Calculate the number of commands
    while (command_count < MAX_COMMANDS && commands[command_count][0] != '\0') {
        command_count++;
    }

    for (int i = 0; i < command_count; i++) {
        if (zephyr_wpa_cli_cmd_v(commands[i]) < 0) {
            wpa_printf(MSG_ERROR, "Failed to execute wpa_cli command: %s", commands[i]);
            status = false;
            break;
        }
    }

    return status;
}

int add_command(char commands[][MAX_COMMAND_LENGTH], const char *format, ...) {
    int command_count = 0;

    // Find the current number of commands
    while (command_count < MAX_COMMANDS && commands[command_count][0] != '\0') {
        command_count++;
    }

    if (command_count >= MAX_COMMANDS) {
        wpa_printf(MSG_ERROR, "Command limit reached");
        return command_count;
    }

    va_list args;
    va_start(args, format);

    // Format the string
    vsnprintf(commands[command_count], MAX_COMMAND_LENGTH, format, args);
    va_end(args);

    return command_count + 1;
}

void example_usage() {
    char commands[MAX_COMMANDS][MAX_COMMAND_LENGTH] = {0};

    if (params->sae_password) {
        add_command(commands, "set_network %d sae_password \"%s\"", network_id, params->sae_password);
    }

    add_command(commands, "set_network %d key_mgmt SAE", network_id);

    if (params->security == WIFI_SECURITY_TYPE_SAE_H2E ||
        params->security == WIFI_SECURITY_TYPE_SAE_AUTO) {
        add_command(commands, "set sae_pwe %d", sae_pwe_value);
    }

    if (!execute_wpa_cli_commands(commands)) {
        goto out;
    }

out:
    // Handle the cleanup or error handling here
}
```

Reply:
I just gave prompt to my copilot to rewrite, so, ignore the details, that was just demonstrate the idea.

Reply:
Sure, I guessed that was the case.